utoipa-gen/src/path/response.rs

```diff
@@ -285,12 +273,12 @@ impl Parse for Header<'_> {
 
 impl ToTokens for Header<'_> {
     fn to_tokens(&self, tokens: &mut TokenStream2) {
-        if let Some(ref header_type) = self.value_type {
+        if let Some(header_type) = &self.value_type {
             // header property with custom type
-            let header_type = Property::new(header_type.is_array, &header_type.ty);
+            let header_type_property = Property::new(header_type.clone());
```

Comment:
This clone could probably be removed if the type was reference or Cow. :) See above :point_up: